modules/Battle.py

```diff
@@ -0,0 +1,834 @@
+        case _:
+            move_menu_task = None
+    move_menu = (move_menu_task != {} and move_menu_task['isActive'])
+
+
+    if move_menu:
+        return "MOVE_MENU"
+    elif stop_learn_move_yes_no:
+        return "STOP_LEARNING"
+    elif learn_move_yes_no:
+        return "LEARN_YN"
+    else:
+        return "NO"
+
+
+def HandleEvolutionScene():
+    """
+    Stops evolution if configured to do so, otherwise mashes A
+    """
+    if config_battle['stop_evolution']:
+        PressButton(['B'])
+    else:
+        PressButton(['A'])
+
+
+def GetCurrentBattler() -> list:
+    """
+    Determines which pokemon is battling
+    """
+    match mGBA.game:
+        case 'Pokémon Ruby' | 'Pokémon Sapphire':
+            # this tells us which pokemon from our party are battling. 0 represents a pokemon in the party who isn't battling, and also the pokemon at index 0 :(
+            battler_indices = [int.from_bytes(ReadSymbol('gBattlerPartyIndexes', size=12)[2*i:2*i+2], 'little') for i in range(len(GetParty()))]
+            # this tells us how many pokemon are battling (2 for single battle, 4 for double) which allows us to get the pokemon info for the party members currently participating in the battle
+            num_battlers = ReadSymbol('gBattlersCount', size=1)[0]
+            # If we only have one party member, it's obviously the current battler so don't do the calcs for who's in battle
+            if len(GetParty()) == 1:
+                return GetParty()
+            current_battlers = [GetParty()[battler_indices[i]] for i in range(len(battler_indices)//num_battlers)]
+            return current_battlers
+        case 'Pokémon Emerald' | 'Pokémon FireRed' | 'Pokémon LeafGreen':
+            # this tells us which pokemon from our party are battling. 0 represents a pokemon in the party who isn't battling, and also the pokemon at index 0 :(
+            battler_indices = [int.from_bytes(ReadSymbol('gBattlerPartyIndexes', size=12)[2*i:2*i+2], 'little') for i in range(len(GetParty()))]
+            # this tells us how many pokemon are battling (2 for single battle, 4 for double) which allows us to get the pokemon info for the party members currently participating in the battle
+            num_battlers = ReadSymbol('gBattlersCount', size=1)[0]
+            # If we only have one party member, it's obviously the current battler so don't do the calcs for who's in battle
+            if len(GetParty()) == 1:
+                return GetParty()
+            current_battlers = [GetParty()[battler_indices[i]] for i in range(len(battler_indices)//num_battlers)]
+            return current_battlers
+
+
+def GetStrongestMove() -> int:
+    """
+    Function that determines the strongest move to use given the current battler and the current
+    """
+    current_battlers = GetCurrentBattler()
+    if len(current_battlers) > 1:
+        console.print("Double battle detected, feature not yet implemented.")
```

Comment:
While setting up a team of Zigzagoon, I am consistently hitting this error - `current_battlers` contains 2 of my lead Pokemon (Grovyle) - save state attached, not sure if you can replicate this issue?

[Pokemon - Emerald Version (USA, Europe) - Double Battle SS.zip](https://github.com/40Cakes/pokebot-gen3/files/12734268/Pokemon.-.Emerald.Version.USA.Europe.-.Double.Battle.SS.zip)